### PR TITLE
Adds UI warning for out of sync Airflow workflow

### PR DESCRIPTION
**Pull request by saurav-c**
## Describe your changes and why you are making these changes
This PR adds a warning on the workflow viewing page if there is a potential out of sync conflict between the DAG registered with Airflow vs the latest Aqueduct dag. For now we just tell the user that they must update the Airflow DAG file.

We can add support later for a user to download the most recent Airflow DAG file from the UI in case they misplaced the one returned by the SDK.

This PR also invokes workflow syncing when the `GET` `/workflow` route is invoked. Previously syncing only occurred during the LIST workflows route.

## Related issue number (if any)
ENG 1705

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

This is what the UI shows when DAGs are out of sync.
Screen Shot 2022-09-20 at 11 01 00 AM

### CODE REVIEW

**Comment by saurav-c** [comment]
cc @vsreekanti for UI changes

File: src/ui/common/src/components/workflows/workflowHeader.tsx [hunk 1]
```diff
@@ -93,6 +94,20 @@ const WorkflowHeader: React.FC<Props> = ({ user, workflowDag, workflowId }) => {
     );
   }
 
+  let airflowUpdateWarning = null;
+  if (workflowDag.engine_config.type == EngineType.Airflow 
+    && !workflowDag.engine_config.airflow_config?.matches_airflow) {
+      airflowUpdateWarning = (
+        <Box>
```
Comment (saurav-c) [inline src/ui/common/src/components/workflows/workflowHeader.tsx:101]: @agiron123 @vsreekanti What's the best way to make this box's width smaller. It takes up almost the entire page right now.

Comment (vsreekanti) [reply src/ui/common/src/components/workflows/workflowHeader.tsx]: I would just set a max width in pixels on the box.

**Comment by vsreekanti** [comment]
Is it just a coincidence that the whole workflow is marked as failed in your screenshot, @saurav-c? The warning looks good to me.

**Comment by saurav-c** [comment]
@vsreekanti Yup, the failure is a coincident

File: src/ui/common/src/components/workflows/workflowHeader.tsx (hunk 1 above)
Comment (agiron123) [reply src/ui/common/src/components/workflows/workflowHeader.tsx]: Yeah, something like maxWidth="800px" would be good to have here. @saurav-c

File: src/ui/common/src/components/workflows/workflowHeader.tsx [hunk 2]
```diff
@@ -254,6 +271,8 @@ const WorkflowHeader: React.FC<Props> = ({ user, workflowDag, workflowId }) => {
         {runWorkflowDialog}
       </Box>
 
+      {airflowUpdateWarning}
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/workflowHeader.tsx:274]: small nit here, I prefer to conditionally render things in a case like this, so we know when the warning will show from reading the render function.

You can do something like:
```
const showAirflowUpdateWarning = workflowDag.engine_config.type == EngineType.Airflow &&
    !workflowDag.engine_config.airflow_config?.matches_airflow;
```

Then, inside render:
```
{showAirflowUpdateWarning && airflowUpdateWarning}
```

**Review by agiron123 — APPROVED** [review]
Frontend stuff looks good. Left a small comment, but that can be resolved before you merge in.

**Review by cw75 — APPROVED** [review]

**Pull request merged**